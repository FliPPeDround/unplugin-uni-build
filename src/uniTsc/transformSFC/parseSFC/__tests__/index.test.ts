import { describe, expect, test } from 'vitest'
import { parseSFC } from '..'

const files = import.meta.glob('./fixtures/*.vue', {
  eager: true,
  as: 'raw',
})

describe('fixtures', async () => {
  console.log(files)
  for (const [id, code] of Object.entries(files)) {
    console.log(code)
    test(id.replace(/\\/g, '/'), () => {
      const exec = () => parseSFC(code)
      if (id.includes('error'))
        expect(exec).toThrowErrorMatchingSnapshot()

      else
        expect(exec()).toMatchSnapshot()
    })
  }
})
