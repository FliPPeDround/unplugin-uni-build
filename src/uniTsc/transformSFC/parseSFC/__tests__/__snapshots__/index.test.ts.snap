// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > ./fixtures/base.vue 1`] = `
{
  "script": undefined,
  "scriptSetup": undefined,
  "styles": [],
  "template": {
    "children": [
      {
        "content": "  ",
        "loc": {
          "end": {
            "column": 3,
            "line": 6,
            "offset": 57,
          },
          "source": "
  ",
          "start": {
            "column": 11,
            "line": 5,
            "offset": 54,
          },
        },
        "type": 2,
      },
      {
        "content": {
          "constType": 0,
          "content": "foo",
          "isStatic": false,
          "loc": {
            "end": {
              "column": 9,
              "line": 6,
              "offset": 63,
            },
            "source": "foo",
            "start": {
              "column": 6,
              "line": 6,
              "offset": 60,
            },
          },
          "type": 4,
        },
        "loc": {
          "end": {
            "column": 12,
            "line": 6,
            "offset": 66,
          },
          "source": "{{ foo }}",
          "start": {
            "column": 3,
            "line": 6,
            "offset": 57,
          },
        },
        "type": 5,
      },
      {
        "content": "
",
        "loc": {
          "end": {
            "column": 1,
            "line": 7,
            "offset": 67,
          },
          "source": "
",
          "start": {
            "column": 12,
            "line": 6,
            "offset": 66,
          },
        },
        "type": 2,
      },
    ],
    "codegenNode": undefined,
    "isSelfClosing": false,
    "loc": {
      "end": {
        "column": 12,
        "line": 7,
        "offset": 78,
      },
      "source": "<template>
  {{ foo }}
</template>",
      "start": {
        "column": 1,
        "line": 5,
        "offset": 44,
      },
    },
    "ns": 0,
    "props": [],
    "tag": "template",
    "tagType": 0,
    "type": 1,
  },
}
`;

exports[`fixtures > ./fixtures/baseTs.vue 1`] = `
{
  "script": undefined,
  "scriptSetup": "
const foo = 'foo'
",
  "styles": [],
  "template": {
    "children": [
      {
        "content": "  ",
        "loc": {
          "end": {
            "column": 3,
            "line": 6,
            "offset": 67,
          },
          "source": "
  ",
          "start": {
            "column": 11,
            "line": 5,
            "offset": 64,
          },
        },
        "type": 2,
      },
      {
        "children": [
          {
            "content": "    ",
            "loc": {
              "end": {
                "column": 5,
                "line": 7,
                "offset": 77,
              },
              "source": "
    ",
              "start": {
                "column": 8,
                "line": 6,
                "offset": 72,
              },
            },
            "type": 2,
          },
          {
            "content": {
              "constType": 0,
              "content": "foo",
              "isStatic": false,
              "loc": {
                "end": {
                  "column": 11,
                  "line": 7,
                  "offset": 83,
                },
                "source": "foo",
                "start": {
                  "column": 8,
                  "line": 7,
                  "offset": 80,
                },
              },
              "type": 4,
            },
            "loc": {
              "end": {
                "column": 14,
                "line": 7,
                "offset": 86,
              },
              "source": "{{ foo }}",
              "start": {
                "column": 5,
                "line": 7,
                "offset": 77,
              },
            },
            "type": 5,
          },
          {
            "content": "
  ",
            "loc": {
              "end": {
                "column": 3,
                "line": 8,
                "offset": 89,
              },
              "source": "
  ",
              "start": {
                "column": 14,
                "line": 7,
                "offset": 86,
              },
            },
            "type": 2,
          },
        ],
        "codegenNode": undefined,
        "isSelfClosing": false,
        "loc": {
          "end": {
            "column": 9,
            "line": 8,
            "offset": 95,
          },
          "source": "<div>
    {{ foo }}
  </div>",
          "start": {
            "column": 3,
            "line": 6,
            "offset": 67,
          },
        },
        "ns": 0,
        "props": [],
        "tag": "div",
        "tagType": 0,
        "type": 1,
      },
      {
        "content": "
",
        "loc": {
          "end": {
            "column": 1,
            "line": 9,
            "offset": 96,
          },
          "source": "
",
          "start": {
            "column": 9,
            "line": 8,
            "offset": 95,
          },
        },
        "type": 2,
      },
    ],
    "codegenNode": undefined,
    "isSelfClosing": false,
    "loc": {
      "end": {
        "column": 12,
        "line": 9,
        "offset": 107,
      },
      "source": "<template>
  <div>
    {{ foo }}
  </div>
</template>",
      "start": {
        "column": 1,
        "line": 5,
        "offset": 54,
      },
    },
    "ns": 0,
    "props": [],
    "tag": "template",
    "tagType": 0,
    "type": 1,
  },
}
`;
